src/main/java/thePackmaster/patches/MetricsPatches.java

```diff
@@ -0,0 +1,110 @@
+            if (AbstractDungeon.player.chosenClass == THE_PACKMASTER) {
+                httpRequest.setMethod("PUT");
+            }
+        }
+
+        private static class Locator extends SpireInsertLocator {
+            @Override
+            public int[] Locate(CtBehavior method) throws Exception {
+                Matcher matcher = new Matcher.MethodCallMatcher(Net.HttpRequest.class, "setContent");
+                return LineFinder.findInOrder(method, matcher);
+            }
+        }
+    }
+
+
+    @SpirePatch(clz = GameOverScreen.class, method = "shouldUploadMetricData")
+    public static class ShouldUploadMetricData {
+        @SpirePostfixPatch
+        public static boolean patch(boolean returnValue) {
+            if (AbstractDungeon.player.chosenClass == THE_PACKMASTER) {
+                returnValue = Settings.UPLOAD_DATA;
+            }
+            return returnValue;
+        }
+
+    }
+
+    @SpirePatch(clz = Metrics.class, method = "run")
+    public static class RunPatch {
+        @SpirePostfixPatch
+        public static void patch(Metrics metrics) {
+            if (metrics.type == Metrics.MetricRequestType.UPLOAD_METRICS && AbstractDungeon.player.chosenClass == THE_PACKMASTER) {
+                try {
+                    Method addDataMethod = Metrics.class.getDeclaredMethod("addData", Object.class, Object.class);
+                    addDataMethod.setAccessible(true);
+                    addDataMethod.invoke(metrics, "pmversion", findTheModVersion());
+                    addDataMethod.invoke(metrics, "currentPacks", SpireAnniversary5Mod.currentPoolPacks.stream().map(p -> p.packID).collect(Collectors.joining(",")));
+
+                    String filteredPacks = SpireAnniversary5Mod.unfilteredAllPacks.stream()
+                            .filter(p -> !SpireAnniversary5Mod.allPacks.contains(p))
+                            .map(p -> p.packID)
+                            .collect(Collectors.joining(","));
+                    addDataMethod.invoke(metrics, "filteredPacks", filteredPacks);
+
+                    //Pack choices at the beginning formatted as Pack1|*Pack2|Pack3,Pack1... * signifying the selected pack
+                    addDataMethod.invoke(metrics, "packChoices", String.join(",", packChoices));
+                    boolean draftEnabled = SpireAnniversary5Mod.getCustomDraftEnabled();
+                    addDataMethod.invoke(metrics, "customDraftEnabled", draftEnabled);
+                    if(draftEnabled)
+                        addDataMethod.invoke(metrics, "customDraftConfiguration", SpireAnniversary5Mod.modConfig.getString("PackmasterCustomDraftSelection"));
+
+                    addDataMethod.invoke(metrics, "pickedHat", SpireAnniversary5Mod.getLastPickedHatID());
+                    addDataMethod.invoke(metrics, "enabledExpansionPacks", Loader.isModLoaded("expansionPacks")); //TODO: Actually use the correct field once both branches are merged
+
```

Comment:
Another data point that would be interesting: whether packs added by mods other than Packmaster and its expansions exist. In other words, how often do people actually run stuff like ocean's mod alongside Packmaster (whether or not they actually use the new packs).

We should be able to tell that by looking at `SpireAnniversary5Mod.unfilteredAllPacks` and checking the namespace that the packs come from. In fact we might be able to just use `WhatMod.findModID`:
```
String expansionModID = "expansionPacks"; // We should probably put this in the main mod file alongside modID, if it's going to be used in multiple places
List<String> packmasterModIDs = Arrays.asList(SpireAnniversary5Mod.modID, expansionModID);
String otherModPackIDs = SpireAnniversary5Mod.unfilteredAllPacks.stream()
		.filter(p -> !packmasterModIDs.contains(WhatMod.findModID(p.getClass())))
		.map(p -> p.packID)
		.collect(Collectors.joining(","));
SpireAnniversary5Mod.logger.info("Packs from other mods: " + otherModPackIDs);
```

I tried that out with ocean's mod running and got `clockworkchar:ToolsPack` as the output.

Reply:
I think a simpler solution would be to just check for anniv5: modid prefix for the pack.